src/services/message.service.ts

```diff
@@ -31,6 +31,46 @@ export class MessageService {
     this._sendConstructedReport(report, { files: message.attachments.map((e) => e.url) });
   }
 
+  async sendMessage(
+    message: IMessage,
+    content: string,
+    reply: boolean,
+    argHeader?: string,
+    argFooter?: string,
+    argBreakChar?: string
+  ): Promise<Boolean> {
+    const header = argHeader ? argHeader + '\n' : '';
+    const footer = argFooter ? '\n' + argFooter : '';
+    const breakChar = argBreakChar || '';
+
+    const replyLength = reply ? ('<@' + message.member?.id + '>,  ').length : 0;
+    const miscLength = header.length + footer.length;
+
+    let messagesToSend: string[] = [header + content + footer];
+
+    if (content.length + miscLength + replyLength > Constants.MaxMessageLength) {
+      if (!breakChar) {
+        return this._failedToSend(message, `Message was over Discords character cap.`);
+      }
+
+      messagesToSend = this._splitMessage(content, breakChar, replyLength + miscLength);
+      if (!this._testMessages(messagesToSend, content.length, replyLength)) {
+        return this._failedToSend(message, `Failed to split the message up within the rule set.`);
+      }
+      messagesToSend = messagesToSend.map((message) => header + message + footer);
+    }
+
+    if (reply) {
+      message.reply(messagesToSend.shift() || '');
+    }
+
+    messagesToSend.forEach((val) => {
+      message.channel.send(val);
```

Comment:
iirc `.send()` is a promise. can we use a regular for loop and await each message send? if any fail, we should let the client know via exception

Reply:
instead, since you already have a map. just
```
await Promise.all(messagesToSend.map((val) => {
    message.channel.send(val);
});
```